docs/content/post/developer-documentation.md

```diff
@@ -0,0 +1,310 @@
+
+Use `hack/build-go.sh` to create a binary of the egress router.
+
+Create a config file in json format called egress-router.conf for the CNI
+ plugin, such as:
+
+```json
+{
+   "cniVersion":"0.4.0",
+   "name":"egress-router",
+   "type":"egress-router",
+   "ip":{
+      "addresses":[
+         "192.168.10.99/24"
+      ],
+      "destinations":[
+         "10.0.3.0/32"
+      ],
+      "gateway":"192.168.10.254"
+   }
+}
+```
+
+Then you can call cnitool in order to test your configuration, and it'll output
+a nice json back.
+
+```bash
+#!/bin/bash
+
+sudo ip netns add testing
+sudo CNI_PATH=egress-router-cni/bin NETCONFPATH=. cnitool add egress-router /var/run/netns/testing
+```
+
+```bash
+{
+    "cniVersion": "0.4.0",
+    "interfaces": [
+        {
+            "name": "eth0",
+            "mac": "1a:9b:80:a1:1e:99",
+            "sandbox": "/var/run/netns/testing"
+        }
+    ],
+    "ips": [
+        {
+            "version": "4",
+            "interface": 0,
+            "address": "192.168.10.99/24",
+            "gateway": "192.168.10.254"
+        }
+    ],
+    "dns": {}
+```
+
+Now we can just enter that namespace and check that the macvlan interface and
+the iptables rules were applied.
+
+```bash
+$ sudo ip netns exec testing ip a
+1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+2: eth0@if6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
+    link/ether 1a:9b:80:a1:1e:99 brd ff:ff:ff:ff:ff:ff link-netnsid 0
+    inet 192.168.10.99/24 brd 192.168.10.255 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::189b:80ff:fea1:1e99/64 scope link
+       valid_lft forever preferred_lft forever
+```
+
+```bash
+$ sudo ip netns exec testing iptables -t nat -L
+Chain PREROUTING (policy ACCEPT)
+target     prot opt source               destination
+DNAT       all  --  anywhere             anywhere             to:10.0.3.0
+
+Chain INPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain OUTPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+SNAT       all  --  anywhere             anywhere             to:192.168.10.99
+
+```
+
+## Using quay.io/openshift image
+
+In order to deploy this into a cluster, you'll have to either compile the
+included `Dockerfile` or rely into the preexising images that are shipped into
+openshift's quay
+(https://quay.io/repository/openshift/origin-egress-router-cni).
+
+If deploying an openshift cluster, we're already deploying this image alongside
+CNO's multus' daemonsets, so you've got nothing to do. Only if you want to
+change the default image to yours, you'll have to override the
+`EGRESS_ROUTER_CNI_IMAGE` value with the reference to your image within
+`env.sh` file for CNO's `hack/run-locally`.
+
```

Comment:
nit: the wording here is a bit awk to me, Maybe something like 
"By default the egress router image is deployed within the CNO's multus daemonsets for all Openshift clusters.  However, If a custom image is required the `EGRESS_ROUTER_CNI_IMAGE` value can be specified in the 
`env.sh` file for CNO's `hack/run-locally`.